source/parser.hera

```diff
@@ -2970,10 +2969,10 @@ PrivateIdentifier
 # NOTE: Merged into MethodDefinition
 
 # NOTE: Allow arbitrary whitespace before regular assignment,
-# but require nonnewline whitespace before operator assignment.
+# but only allow non-newline whitespace before operator assignment.
 WAssignmentOp
   __ AssignmentOp
-  _ OperatorAssignmentOp
+  _? OperatorAssignmentOp
```

Comment:
I think this change allows something like `x[1]xor= true` or `x[1]min= x[0]` which doesn't look great... but I guess it's OK, looking at the new test case. Certainly can't mean anything else.